Alicerce/Sources/Network/Parse.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Parse.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 06/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import UIKit
+
+public enum Parse {
+    enum Error: Swift.Error {
+        case serialization(String)
+        case mapping(MappableError)
+        case other(Swift.Error)
+    }
+
+    /// Converts raw data into a `T` object
+    ///
+    /// - Parameter data: Raw data
+    /// - Returns: A `T` filled with values from the raw Data
+    /// - Throws: 
+    /// A Parse.Error that can be of type: 
+    ///   - `serialization` if json serialization failed
+    ///   - `mapping` if model mapping failed
+    ///   - `other` an unknown error
+    static func json<T: Mappable>(data: Data) throws -> T {
+
+        let decodedObject: T
+
+        do {
+            let json = try JSONSerialization.jsonObject(with: data, options: [])
+
+            decodedObject = try T.model(from: json)
```

Comment:
I think we can return directly from inside the `do..catch` block, so no need for the external var at the beginning of the func 😉